scipy/special/add_newdocs.py

```diff
@@ -6348,16 +6544,206 @@ def add_newdoc(place, name, doc):
     equality between an empirical and a theoretical distribution. It
     is equal to the probability that the maximum difference between a
     theoretical distribution and an empirical one based on `n` samples
-    is greater than e.
+    is greater than d.
+
+    Parameters
+    ----------
+    n : int
+      Number of samples
+    d : float array_like
+      Deviation between the ECDF and the CDF
+
+    Returns
+    -------
+    float
+        The value(s) of smirnov(n, d), Prob(Dn >= d)
+
+    See Also
+    --------
+    smirnovc : The CDF for the distribution
+    smirnovi : The Inverse Survival Function for the distribution
+    smirnovp : The derivative of the Survival Function for the distribution.
+    scipy.stats.ksone : Provides the functionality as a continuous distribution
+    kolmogorov, kolmogi, kolmogc, kolmogci, kolmogp : Functions for the two-sided distribution
+
+    Examples
+    --------
+    >>> from scipy.special import smirnov
+
+    Show the probability of a gap at least as big as 0, 0.5 and 1.0 for a sample of size 5
+
+    >>> smirnov(5, [0, 0.5, 1.0])
+    array([ 1.   ,  0.056,  0.   ])
+
+    Compare a sample of size 5 drawn from a Normal(0.5, 1) distribution against
+    a target Normal(0, 1) CDF.
+
+    >>> from scipy.stats import norm
+    >>> n = 5
+    >>> gendist = norm(0.5,1)        # Normal distribution, mean 0.5, stddev 1
+    >>> np.random.seed(seed=233423)  # Set the seed for reproducibility
+    >>> x = np.sort(gendist.rvs(size=n))
+    >>> x
+    array([-0.20946287,  0.71688765,  0.95164151,  1.44590852,  3.08880533])
+    >>> target = norm(0, 1)
+    >>> cdfs = target.cdf(x)
+    >>> cdfs
+    array([ 0.41704346,  0.76327829,  0.82936059,  0.92589857,  0.99899518])
+    # Construct the Empirical CDF and the K-S statistics (Dn+, Dn-, Dn)
```

Comment:
Do you compute Dn? Could even be useful to add to the example and compute asymptotic p-value to show the link to the kolmog functions. Though n=5 not ideal for asymptotic result :)